llvm/lib/Target/AMDGPU/SIPreEmitPeephole.cpp

```diff
@@ -299,7 +299,7 @@ bool SIPreEmitPeephole::optimizeSetGPR(MachineInstr &First,
   for (MachineBasicBlock::instr_iterator I = std::next(First.getIterator()),
                                          E = MI.getIterator();
        I != E; ++I) {
-    if (I->isBundle())
+    if (I->isBundle() || I->isDebugInstr())
```

Comment:
Probably should be all meta instructions

Reply:
Done for KILL. This is post-RA so I don't think we'll see IMPLICIT_DEF.

Reply:
IMPLICIT_DEF often survives RA, it is needed to maintain liveness across blocks